feat: add public export

prawwtocol opened the pull request.

prawwtocol committed:
feat: add public export

prawwtocol committed:
fix: correct export_public parameters and improve commit command syntax

prawwtocol committed:
refactor: update _commit_and_push_repo method signature to include pragma directive

zhmurov commented on checker/exporter.py:
```diff
@@ -568,6 +572,61 @@ def _copy_files_with_config(
+
+        print_info("* committing...")
+        r = subprocess.run(
+            ["git", "commit", "--all", "-m", message],
```
``git commit`` does not have option ``--all``, ``git add`` has

prawwtocol committed:
fix: remove shell=True from git commit subprocess call

shell=True with list argument causes incorrect behavior - the list is not
passed to git but to shell. Also added handling for 'nothing to commit' case.

prawwtocol commented:
дока по public export, нужно куда-то добавить?

zhmurov committed:
Merge branch 'main' into feat/public-export
